hkdf/src/main/java/keywhiz/hkdf/Hkdf.java

```diff
@@ -148,10 +149,11 @@ public SecretKey extract(@Nullable SecretKey salt, byte[] ikm) {
     Mac mac = initMac(key);
     for (int roundNum = 1; roundNum <= n; roundNum++) {
       mac.reset();
-      mac.update(hashRound);
-      mac.update(info);
-      mac.update((byte) roundNum);
-      hashRound = mac.doFinal();
+      ByteBuffer t = ByteBuffer.allocate(hashRound.length + info.length + 1);
```

Comment:
note: it might look inefficient to perform allocations inside a loop, but we only ever loop once.